src/core/api/retention.go

```diff
@@ -175,13 +178,13 @@ func (r *RetentionAPI) CreateRetention() {
 			r.SendBadRequestError(err)
 		}
 		if proj == nil {
-			r.SendBadRequestError(fmt.Errorf("Invalid Project id %d", p.Scope.Reference))
+			r.SendBadRequestError(fmt.Errorf("invalid Project id %d", p.Scope.Reference))
 		}
 	default:
 		r.SendBadRequestError(fmt.Errorf("scope %s is not support", p.Scope.Level))
 		return
 	}
-	if err = retentionController.CreateRetention(p); err != nil {
+	if _, err = retentionController.CreateRetention(p); err != nil {
```

Comment:
We should return the ID just like https://github.com/goharbor/harbor/blob/master/src/core/api/replication_policy.go#L104